### PR TITLE
Add detection NN example

### PR DESCRIPTION
**Pull request by szabi-luxonis**
Added getLayerInt32 for NN results, used by deeplabv3 for example
Added mobilenet device side decoding example
Added tiny yolo v3 device side decoding example
Improved Detection network API

Fix FW bugs: https://gitlab.com/luxonis/depthai/-/merge_requests/104

### CODE REVIEW

File: include/depthai/pipeline/node/DetectionNetwork.hpp [hunk 1]
```diff
@@ -26,6 +26,7 @@ class DetectionNetwork : public NeuralNetwork {
    public:
     Input input{*this, "in", Input::Type::SReceiver, {{DatatypeEnum::ImgFrame, true}}};
     Output out{*this, "out", Output::Type::MSender, {{DatatypeEnum::Buffer, false}}};
+    Output passthrough{*this, "passthrough", Output::Type::MSender, {{DatatypeEnum::Buffer, true}}};
```
Comment (themarpe) [inline include/depthai/pipeline/node/DetectionNetwork.hpp:29]: `DatatypeEnum::ImgFrame, true` same as input (if it really does passthrough input only)

File: examples/src/camera_mobilenet_example.cpp [hunk 2]
```diff
@@ -30,7 +31,11 @@ dai::Pipeline createNNPipeline(std::string nnPath){
 
     // Link plugins CAM -> NN -> XLINK
     colorCam->preview.link(nn1->input);
-    colorCam->preview.link(xlinkOut->input);
+    if (syncNN)
```
Comment (themarpe) [inline examples/src/camera_mobilenet_example.cpp:34]: Inline or `{}`

File: include/depthai/pipeline/datatype/ImgDetections.hpp [hunk 3]
```diff
@@ -18,8 +18,7 @@ class ImgDetections : public Buffer {
     explicit ImgDetections(std::shared_ptr<RawImgDetections> ptr);
     virtual ~ImgDetections() = default;
 
-    // reference
-    std::vector<ImgDetection>& detections;
+    std::vector<ImgDetection> getDetections(void) const;
```
Comment (themarpe) [inline include/depthai/pipeline/datatype/ImgDetections.hpp:21]:
```
std::vector<ImgDetection> getDetections() const;
```

**Review by themarpe — CHANGES_REQUESTED** [review]
LGTM apart from the requested changes.
Rationally behind the ImgDetections move from reference to getter?
And in case it remains, add a setter as well to be able to create ImgDetections message and send it to the device

File: include/depthai/pipeline/node/DetectionNetwork.hpp [hunk 4]
```diff
@@ -26,6 +26,7 @@ class DetectionNetwork : public NeuralNetwork {
    public:
     Input input{*this, "in", Input::Type::SReceiver, {{DatatypeEnum::ImgFrame, true}}};
     Output out{*this, "out", Output::Type::MSender, {{DatatypeEnum::Buffer, false}}};
```
Comment (themarpe) [inline include/depthai/pipeline/node/DetectionNetwork.hpp:28]: This should be `DatatypeEnum::ImgDetections`

File: include/depthai/pipeline/Pipeline.hpp [hunk 5]
```diff
@@ -147,6 +147,10 @@ class Pipeline {
     const AssetManager& getAssetManager() const {
         return impl()->assetManager;
     }
+
+    void setPipelineOpenVINOVersion(OpenVINO::Version version) {
```
Comment (themarpe) [inline include/depthai/pipeline/Pipeline.hpp:151]: I'd rather go with `setOpenVINOVersion` as Pipeline is infered context.

**Comment by szabi-luxonis** [comment]
> LGTM apart from the requested changes.
> Rationally behind the ImgDetections move from reference to getter?
> And in case it remains, add a setter as well to be able to create ImgDetections message and send it to the device

Switched back to reference, fixed setNumFramePools for detection NN, fixed exit on `q` press (waitkey) in all examples.

**Review by themarpe — APPROVED** [review]

**Pull request merged**